src/chords.py

```diff
@@ -0,0 +1,164 @@
+    """
+    # mido_obj = midipiece_to_midifile(piece)
+    # define comparator
+    setattr(Note, "__lt__", lambda self, other: self.end <= other.end)
+    pq = queue.PriorityQueue()
+    start = 0
+    count = 0
+    chords = []
+    prev_chord = None
+    midi_data = piece.df
+    piece = piece.to_midi()
+    midi_data["start"] = midi_data.apply(start_to_ticks, axis=1, args=(piece,))
+    midi_data["end"] = midi_data.apply(end_to_ticks, axis=1, args=(piece,))
+    notes_midi = np.array(midi_data[["velocity", "pitch", "start", "end"]])
+
+    for note in notes_midi:
+        note = Note(*note)
+        pq.put(note)
+        end = note.end
+        notes = [note for note in pq.queue]
+        # Pop notes which ended before current note started
+        if pq.empty():
+            continue
+        while note.start > pq.queue[0].end:
+            pq.get()
+            start = pq.queue[0].start
+        # Try to recognize chord from list of notes
+        chord = chorder.Dechorder.get_chord_quality(notes, start, end)
+        if chord[0].is_complete():
+            # Check if the chord has changed
+            if prev_chord != chord[0]:
+                prev_chord = chord[0]
+                chords.append([chord[0].__str__(), start, end])
+                count += 1
+    return chords
+
+
+def plot_chords_vs_time(chords, prettyMIDIpiece, title, timeframe=None):
+    """
+    Plots the number of chords played over time.
+
+    Parameters:
+        chords (list): A list of detected chords in the format [chord_name, start_time, end_time].
+        prettyMIDIpiece (PrettyMIDI): A PrettyMIDI object representing the MIDI data.
+        title (str): The title of the plot.
+        timeframe (int, optional): The time interval (in seconds) for grouping chords. If not provided,
+                                   it will be automatically calculated based on the recording duration.
+
+    Returns:
+        None
+    """
+    chords = pd.DataFrame(chords, columns=["chord", "start", "end"])
+    chords["end"] = chords.apply(end_to_time, axis=1, args=(prettyMIDIpiece,))
+
+    recording_duration_seconds = chords["end"].max()
+    # Determine the timeframe for grouping chords based on the recording duration
+    if timeframe is None:
+        if recording_duration_seconds > 120:
+            # Calculate the number of notes played per minute
+            timeframe = 60
+            x_label = "Minutes"
+        else:
+            timeframe = 1
+            x_label = "Seconds"
+    else:
+        x_label = "{:d} seconds".format(timeframe)
+    # Group chords based on the specified timeframe and count the number of chords in each group
+    chords["timeframe"] = chords["end"] // timeframe
+    chords_count = chords.groupby(["timeframe"]).size()
+    plot(
+        x=chords_count.index, y=chords_count.values, xlabel=x_label, ylabel="Chords per {:s}".format(x_label.lower()), title=title
+    )
+
+
+def process_dataset(dataset, csv_file_path):
```

Comment:
```suggestion
def process_dataset(dataset: Dataset) -> pd.DataFrame:
```

I think it's a good practice to separate data processing logic from any data presentation/visualization procedures, so the csv creation could be moved to the main part:

```python
df = process_dataset(dataset)
df.to_csv(csv_file_path)
```

I also included type hints in my suggestion - these are very helpful in maintaining readability :)